### ISSUE TITLE
Enabling removed functionality of required_fields and required_annotations

### ISSUE THREAD

**Issue by MrThearMan** [MrThearMan/graphene-django-query-optimizer#85]
_Originally posted by @rsomani95 in https://github.com/MrThearMan/graphene-django-query-optimizer/issues/77#issuecomment-2008278737_

...

However, I have a few concerns (questions, really) around how to reformulate what `required_fields` and `annotated_fields` enabled for me before. 

---

### `required_fields`

In this simplified example, I have a computed property defined on the Django model directly, like so:

```python
class RootModel:
    bucket = models.CharField()
    key = models.CharField()

    @property
    def params(self):
        return {"Bucket": self.bucket, "Key": self.key}

    @property
    def url(self):
        cache_key = self.cache_key
        params = self.signed_url_params

        # Save to cache if not in cache, else fetch from cache
```

For which the `@required_fields` worked really well. I simply added the annotation and it did everything in one query.
```python
class UrlMixin(DjangoObjectType):
    @required_fields("key", "bucket")
    def resolve_signed_url(root: RootModel, info):
        return root.url
```

I'm not sure how to do this with `AnnotatedField` though.

---

### `annotated_fields`

An `AnnotatedField` currently only takes a single expression, whereas previously we could pass multiple `**kwargs` to `@required_annotations`.

Earlier, I had an annotated field like so:

```python
class Blah(DjangoObjectType):
    @required_annotations(
        in_time_seconds=Cast(F("in_time"), models.FloatField()) / F("in_time_base"),
        out_time_seconds=Cast(F("out_time"), models.FloatField()) / F("out_time_base"),
        duration=F("out_time_seconds") - F("in_time_seconds"),
    )
    def resolve_duration(root: TModel, info: GQLInfo):
        return root.duration
```

which I think isn't possible with this current design (I don't think the above can be formulated into a single `expression`). Maybe `AnnotatedField` can accept either one, or a list of expressions, and work accordingly. Thoughts?

**Comment by MrThearMan** [comment]
To address the missing funtionality once enabled with `required_fields`, I've added a new `MultiField` that accepts multiple field names as an iterable, which will then be added to the queryset's `only`-fields. This does not support related fields (i.e. `relation__field`).

For `annotated_fields` case, I belive that any django expression consisting of multiple annotations _can_ be rewritten as as single annotation. Still, I've added the option to add aliases to the querysets in `AnnotatedField` with the `aliases` parameter. This should make writing the expressions cleaner at the very least.

I also fixed a few bugs related to annotated fields and their naming.

Released in [0.4.1](https://github.com/MrThearMan/graphene-django-query-optimizer/releases/tag/v0.4.1).
